lib/Doctrine/DBAL/DBALException.php

```diff
@@ -277,4 +279,16 @@ public static function typeNotFound($name)
     {
         return new self('Type to be overwritten ' . $name . ' does not exist.');
     }
+
+    public static function typeNotRegistered(Type $type) : self
+    {
+        return new self(sprintf('Type of the class %s@%s is not registered.', get_class($type), spl_object_hash($type)));
+    }
+
+    public static function typeAlreadyRegistered(Type $type) : self
+    {
+        return new self(
+            sprintf('Type of the class %s@%s is already registered.', get_class($type), spl_object_hash($type))
+        );
+    }
```

Comment:
On one hand, not a big fan of making a big exception class that grows with time, on the other hand, if these exceptions are not meant to be caught but to help the user figure out their issue, maybe this is fine?

Reply:
It's only for 2.x.